Project-1/README.md

```diff
@@ -17,48 +17,164 @@ Note: The standard interpretation of the logical symbols - "∨", "∧", "→",
   (b) (A→B)∧(A→¬B)
   (c) (A→(B∨C))∨(C→¬A) 
   (d) ((A→B)∧C)∨(A∧D) 
+  
+  (a) (¬A→B)∨((A∧¬C)→B), Tautology
+  A	B	C	((¬A → B) ∨ ((A ∧ ¬C) → B))
+F	F	F	T
+F	F	T	T
+F	T	F	T
+F	T	T	T
+T	F	F	T
+T	F	T	T
+T	T	F	T
+T	T	T	T
+  (b) (A→B)∧(A→¬B), Contingent
+  A	B	((A → B) ∧ (A → ¬B))
+F	F	T
+F	T	T
+T	F	F
+T	T	F
+  (c) (A→(B∨C))∨(C→¬A), Tautology
+  A	B	C	((A → (B ∨ C)) ∨ (C → ¬A))
+F	F	F	T
+F	F	T	T
+F	T	F	T
+F	T	T	T
+T	F	F	T
+T	F	T	T
+T	T	F	T
+T	T	T	T
+  (d) ((A→B)∧C)∨(A∧D), Contingent
+  A	B	C	D	(((A → B) ∧ C) ∨ (A ∧ D))
+F	F	F	F	F
+F	F	F	T	F
+F	F	T	F	T
+F	F	T	T	T
+F	T	F	F	F
+F	T	F	T	F
+F	T	T	F	T
+F	T	T	T	T
+T	F	F	F	F
+T	F	F	T	T
+T	F	T	F	F
+T	F	T	T	T
+T	T	F	F	F
+T	T	F	T	T
+T	T	T	F	T
+T	T	T	T	T
   ```
 	
 2. A _literal_ is an atomic formula or the negation of an atomic formula. We say a formula is in _conjunctive normal form_ (CNF) if it is the conjunction of the disjunction of literals. Find propositional logic formulas in CNF equivalent to each of the following:
   ```
   (a) (A→B)→C
   (b) (A→(B∨C))∨(C→¬A)
   (c) (¬A∧¬B∧C)∨(¬A∧¬C)∨(B∧C)∨A 
+  
+  (A→B)→C, CNF: (A∨C)∧(¬B∨C)
+  (1) ¬(A→B)∨C
+  (2) ¬(¬A→B)∨C
+  (3) ¬(A→ ¬B)∨C
+  (4) (A∨C)∧ (¬B∨C)
+  
+  (b) (A→(B∨C))∨(C→¬A), CNF: B ∨ ¬B (tautology)
+  
+  (c) (¬A∧¬B∧C)∨(¬A∧¬C)∨(B∧C)∨A, CNF: B ∨ ¬B (tautology) 
+  
   ```
   
 3. Let V be the vocabulary of first-order logic consisting of a binary relation P and a unary relation F. Interpret P(x,y) as “x is a parent of y” and F(x) as “x is female.” Where possible define the following formulas in this vocabulary; where not possible, explain why: 
   
   ```
   (a)  B(x,y) that says that x is a brother of y  
+  
+   B(x,y) <--> ∃z((Pzx ∧ Pzy) ∧ x ≠ y ∧ ~Fx)
+   
   (b)  A(x,y) that says that x is an aunt of y  
-  (c)  C(x,y) that says that x and y are cousins   
+  
+  A(x,y) <--> ∃z((Fx ∧ Pzy ∧ ∃w(Pwx ∧ Pwz ∧ x ≠ z))
+  
+  (c)  C(x,y) that says that x and y are cousins 
+  
+  C(x,y) <--> ∃z∃w∃r(Pzx ∧ Pwy ∧ Prz ∧ Prw ∧ z ≠ w ∧ x ≠ y)
+  
+  
   (d)  O(x) that says that x is an only child  
+  
+  O(x) <--> ∃y∀z(Pyx ∧ (Pyz --> z = x))
+  
   (e)  T(x) that says that x has exactly two brothers 
+  
+  ∃y∃z∃w(¬(w=z)∧¬(x=w)∧¬(x=z)∧P(y,x)∧P(y,z)∧P(y,w)∧¬F(z)∧¬F(w))∧∀r(P(y,r)∧¬F(r)→ r=x ∨ r=z ∨ r=w)
+  
   ```
 
 4. Let V be a vocabulary of the attribute (concept) language with complements (ALC) consisting of a role name "parent_of" and a concept name "Male". Interpret parent_of as "x is a parent of y" and M as "x is male". Where possible define the following formulas in this vocabulary; where not possible, explain why: 
   ```
   (a)  B that says that x is a brother of y
+  
```

Comment:
ALC doesn't have the expressivity to represent cardinalities like >=2. It may not be possible to represent these sentences in ALC...

Reply:
To me, this means that any relation that involves more than two parties to describe (for example, an "aunt" requires three parties - aunt (who has a sister), the sister (who is also a mother), and at least one niece or nephew). 

But then "brothers" confuses me.  On one hand, perhaps only two parties (entities?) are required - two siblings (where, traditionally, one is a male).  However, a third party may be required - a parent conjoining both siblings.  Which is correct?  To define in ALC, do I simply define as preferred (as long as my definition is coherent?) Is this the challenge of the question?